src/scheduler/task.rs

```diff
@@ -356,49 +305,22 @@ impl PriorityTaskQueue {
 	/// Change priority of specific task
 	pub fn set_priority(&mut self, handle: TaskHandle, prio: Priority) -> Result<(), ()> {
 		let i = handle.get_priority().into() as usize;
-		let mut pos = self.queues[i].head.as_mut().ok_or(())?;
-
-		loop {
-			if handle.id == pos.borrow().id {
-				let task = pos.clone();
-
-				// Extract found task from queue and set new priority
-				{
-					let mut borrow = task.borrow_mut();
-
-					let new = borrow.next.as_ref().cloned();
-					if let Some(prev) = borrow.prev.as_mut() {
-						prev.borrow_mut().next = new;
-					}
-
-					let new = borrow.prev.as_ref().cloned();
-					if let Some(next) = borrow.next.as_mut() {
-						next.borrow_mut().prev = new;
-					}
 
-					if borrow.prev.as_mut().is_none() {
-						// Ok, the task is head of the list
-						self.queues[i].head = borrow.next.as_ref().cloned();
-					}
-
-					if borrow.next.as_mut().is_none() {
-						// Ok, the task is tail of the list
-						self.queues[i].tail = borrow.prev.as_ref().cloned();
-					}
-
-					borrow.prio = prio;
-					borrow.next = None;
-					borrow.prev = None;
-				}
+		for (index, current_task) in self.queues[i].iter().enumerate() {
+			// Move the task from its old list to the new list.
+			if handle.id == current_task.borrow().id {
+				// Calling remove is unstable: https://github.com/rust-lang/rust/issues/69210
+				let mut split_list = self.queues[i].split_off(index);
+				let task = split_list.pop_front().ok_or(())?;
+				self.queues[i].append(&mut split_list);
```

Comment:
We might want to extract this code into a function (be it inline or outside this method) for clarity.